aws/resource_aws_lambda_function.go

```diff
@@ -382,6 +382,10 @@ func resourceAwsLambdaFunctionCreate(d *schema.ResourceData, meta interface{}) e
 				log.Printf("[DEBUG] Received %s, retrying CreateFunction", err)
 				return resource.RetryableError(err)
 			}
+			if isAWSErr(err, "InvalidParameterValueException", "Lambda was unable to configure access to your environment variables because the KMS key is invalid for CreateGrant") {
```

Comment:
In a followup commit (so we can release this today), I will also add this retry condition to the `UpdateFunctionConfiguration` call as well to prevent any odd cases with updating IAM roles or KMS backed environment variables.